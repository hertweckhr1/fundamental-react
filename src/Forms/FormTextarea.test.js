import FormTextarea from './FormTextarea';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<Forms />', () => {
    const formTextArea = (
        <FormTextarea className='blue' id='textarea-1'>
                    Pellentesque metus lacus commodo eget justo ut rutrum varius nunc.
        </FormTextarea>
    );

    test('create form text area item', () => {
        // create form set with form inputs
        let component = renderer.create(formTextArea);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the FormTextarea component', () => {
            const element = mount(<FormTextarea data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
