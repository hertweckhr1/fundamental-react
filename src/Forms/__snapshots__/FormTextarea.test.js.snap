// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forms /> create form text area item 1`] = `
<textarea
  className="fd-form__control blue"
  id="textarea-1"
>
  Pellentesque metus lacus commodo eget justo ut rutrum varius nunc.
</textarea>
`;
